contracts/sol6/KyberNetwork.sol

```diff
@@ -574,24 +574,24 @@ contract KyberNetwork is WithdrawableNoModifiers, Utils5, IKyberNetwork, Reentra
         ReservesData memory reservesData = (src == ETH_TOKEN_ADDRESS)
             ? tradeData.ethToToken
             : tradeData.tokenToEth;
-        uint256 callValue;
 
-        for (uint256 i = 0; i < reservesData.addresses.length; i++) {
-            callValue = (src == ETH_TOKEN_ADDRESS) ? reservesData.srcAmounts[i] : 0;
+        uint256 srcDecimals = getUpdateDecimals(src);
```

Comment:
do we need update here?
we have setDecimals in listTokenForReserve

Reply:
If the decimal is already set for this token, the effect (and properly gas consumption) is the same for `getUpdateDecimals` and `getDecimals` right?

Reply:
I would do it differently, not sure if more gas efficient:
```
if (src == ETH_TOKEN_ADDRESS) {
   reservesData = tradeData.ethToToken;
   srcDecimals = ETH_DECIMALS;
   destDecimals = reservesData.decimals;
} else {
   reservesData = tradeData.tokenToEth;
   srcDecimals = reservesData.decimals;
   destDecimals = ETH_DECIMALS;
```

Reply:
actually looking again.
we could be sending correct reserveData from calling function. don't need tradeData

on same lines
can be sending src / dest Decimals.
don't need to query from storage.
And amount can be removed if stack too deep (its being removed in another PR)

Reply:
Updated